astropy/stats/sigma_clipping.py

```diff
@@ -180,6 +180,7 @@ def sigma_clipped_stats(data, mask=None, mask_val=None, sigma=3.0, iters=None):
         image.
     """
 
+    data = np.asanyarray(data)
```

Comment:
Duh, sorry for not looking beyond the line! Since `sigma_clip` uses a masked array internally, it might be an idea to initialise that here already, i.e., `data = np.ma.MaskedArray(data, mask)`. This would allow you to remove the `if mask is not None` clause, and you could use `masked_val` to add to the mask. But perhaps I'm making things unnecessarily complicated...